File: frontend/app/models/task.js
import Model from 'ember-data/model';
import DS from 'ember-data';
import attr from 'ember-data/attr';
import { belongsTo, hasMany } from 'ember-data/relationships';

export default Model.extend({
    task_description: attr('string'),
    planned_date: attr('date'),
    num_of_clients: attr('number'),
    done: attr('boolean', { defaultValue: false }),
    excuted_date: attr('date'),
    clients: hasMany('client'),
    seller: belongsTo('seller'),
    content: belongsTo('content')
});

File: backend/models/task/task.py
from backend import db
from backend.models import BaseColumnsMixin, DictMixin


class Task(db.Model, BaseColumnsMixin, DictMixin):
    __tablename__ = 'task'

    type_of_task = db.Column(db.String(255), nullable=False)
    seller_id = db.Column(db.ForeignKey(
        'seller.id', deferrable=True, initially='DEFERRED', ondelete='CASCADE'), nullable=False, index=True)
    num_of_clients = db.Column(db.Numeric(10), nullable=False, default=0)
    task_description = db.Column(db.Text, nullable=False)
    content_id = db.Column(db.ForeignKey(
        'content.id', deferrable=True, initially='DEFERRED'), nullable=True, index=True)
    medium_id = db.Column(db.ForeignKey(
        'socialnetwork.id', deferrable=True, initially='DEFERRED'), nullable=False, index=True)
    planned_date = db.Column(db.DateTime(True), nullable=False)
    done = db.Column(db.Boolean, nullable=False, default=False)
    excuted_date = db.Column(db.DateTime(True))

    seller = db.relationship('Seller')
    content = db.relationship('Content')
    medium = db.relationship('SocialNetwork')
